ql/src/test/org/apache/hadoop/hive/ql/hooks/TestHiveProtoLoggingHook.java

```diff
@@ -322,4 +338,63 @@ private void assertOtherInfo(HiveHookEventProto event, OtherInfoType key, String
       Assert.assertEquals(value, val);
     }
   }
+
+  private void testTablesWritten(WriteEntity we, boolean isPartitioned) throws Exception {
+    String query = isPartitioned ?
+            "insert into test_partition partition(dt = '20220102', lable = 'test1') values('20220103', 'banana');" :
+            "insert into default.testTable1 values('ab')";
+    HashSet<WriteEntity> tableWritten = new HashSet<>();
+    tableWritten.add(we);
+    QueryState state = new QueryState.Builder().withHiveConf(conf).build();
+    @SuppressWarnings("serial")
+    QueryPlan queryPlan = new QueryPlan(HiveOperation.QUERY) {
+    };
+    queryPlan.setQueryId("test_queryId");
+    queryPlan.setQueryStartTime(1234L);
+    queryPlan.setQueryString(query);
+    queryPlan.setRootTasks(new ArrayList<>());
+    queryPlan.setInputs(new HashSet<>());
+    queryPlan.setOutputs(tableWritten);
+    PerfLogger perf = PerfLogger.getPerfLogger(conf, true);
+    HookContext ctx = new HookContext(queryPlan, state, null, "test_user", "192.168.10.11",
+            "hive_addr", "test_op_id", "test_session_id", "test_thread_id", true, perf, null);
+
+    ctx.setHookType(HookType.PRE_EXEC_HOOK);
+    EventLogger evtLogger = new EventLogger(conf, SystemClock.getInstance());
+    evtLogger.handle(ctx);
+    evtLogger.shutdown();
+
+    HiveHookEventProto event = loadEvent(conf, tmpFolder);
+
+    Assert.assertEquals(EventType.QUERY_SUBMITTED.name(), event.getEventType());
+    Assert.assertEquals(we.getTable().getFullyQualifiedName(), event.getTablesWritten(0));
+  }
+
+  private Table newTable(boolean isPartitioned) {
```

Comment:
would it be better to use builder:
````
TableBuilder builder =
        new TableBuilder().setDbName(DB_NAME).setTableName(tableName).addCol("test_id", "int", "test col id")
            .addCol("test_value", "string", "test col value").setPartCols(partCols)
            .setType(tableType).setLocation(location).setTemporary(true);
    if (tableParams != null) {
      builder.setTableParams(tableParams);
    }
    builder.build(conf);
````

Reply:
WriteEntity constructor only accepts  org.apache.hadoop.hive.ql.metadata.Table . 

If we have to use the the TableBuilder, we will need to import org.apache.hadoop.hive.metastore.api.Table, this will introduce a new package dependency. 

I also found that, the helper methods are available here, we just have move them to commonTestUtils so that other tests can use them. Can we do this in separate jira? 
https://github.com/apache/hive/blob/master/standalone-metastore/metastore-server/src/test/java/org/apache/hadoop/hive/metastore/client/TestAppendPartitions.java#L540

Reply:
sure